pkg/generator/pool_test.go

```diff
@@ -11,17 +13,21 @@ import (
 	"github.com/keep-network/keep-core/pkg/internal/testutils"
 )
 
-// TestGet covers the most basic path - calling `Get()` function multiple times
-// and making sure result is always returned, assuming there are no errors from
-// the persistence layer. The pool size is lower than the number of parameters
-// fetched from the pool so this test also ensures the pool does not stop
-// generating numbers for future `Get()` calls.
-func TestGet(t *testing.T) {
+// TestGetNow covers the most basic path - calling `GetNow()` function multiple
+// times and making sure result is always returned, assuming there are no errors
+// from the persistence layer.
+func TestGetNow(t *testing.T) {
 	pool, scheduler, _ := newTestPool(5)
 	defer scheduler.stop()
 
-	for i := 0; i < 70; i++ {
-		e, err := pool.Get()
+	for {
+		if pool.CurrentSize() == 5 {
+			break
+		}
```

Comment:
```suggestion
		if pool.CurrentSize() == 5 {
			break
		}
		runtime.Gosched()
```

I would yield the processor on each loop execution to make sure we are not stuck too long in the for and the generating goroutine does not get starved.

Reply:
https://github.com/keep-network/keep-core/pull/3207/commits/f44ebc38ff7291cb47df1e3b081337065c840d11